drivers/pcie/CMakeLists.txt

```diff
@@ -1,4 +1,6 @@
-zephyr_sources(pcie.c)
+# Right now only X86 implements required symbols
+# e.g pcie_conf_read and pcie_conf_write
+zephyr_sources_ifdef(CONFIG_X86 pcie.c)
```

Comment:
instead, define per architecture capability, for example ARCH_HAS_PCIE, set this to y for those architectures and here check for 

`zephyr_sources_ifdef(CONFIG_ARCH_HAS_PCIE pcie.c)`

checking for x86 here is not clean.

Reply:
Rather just put `CONFIG_PCIE=y` in x86 boards requiring this? And change this to `zephyr_sources_ifdef(CONFIG_PCIE pcie.c)`.

Reply:
CONFIG_PCIE is needed to build anything in drivers/pcie, doing this will exclude the endpoint drivers from building.

Reply:
Rather than adding architecture capability, which will end up being PCIe RC users setting CONFIG_ARCH_HAS_PCIE, and PCIe EP users not setting the same (which actually sounds confusing), we can go back to having PCIE and PCIE_ENDPOINT completely independent - but this may not be good if in future, we need to reuse some APIs for both. Other way is to let linker handle it, as it's happening right now - I don't see any functionality issue, but I am not sure if it's proper way.

Reply:
@nashif PCI-E as we currently have it implemented is for the host CPU talking to peripherals. Basically part of our "big-core" support package.

The PCIE endpoint stuff is for microcontrollers running *on* a PCI-E peripheral.

They should not be under the same Kconfig, or have one depend on another.

Reply:
not saying they should depend on each, I just do not like having pcie.c depend on X86 as this PR is trying to do, right now the way things are done, everything is under PCIE and that is where things fall apart, to resolve this:

under pci/ add pcie/ and endpoint/ and make cmake for pci/ not depend on any kconfig. CONFIG_PCIE builds pcie/ and CONFIG_PCI_ENDPOINT builds whatever we have in endpoint/

this way, we do not need to depend on X86.